src/site/content/en/blog/reporting-observer/index.md

```diff
@@ -0,0 +1,216 @@
+these different reports to a backend. The Reporting API is a generic framework
+to specify a set of server endpoints to report issues to.
+{% endAside %}
+
+## The API
+
+`ReportingObserver` is not unlike the other "observer" APIs such as
+`IntersectionObserver` and `ResizeObserver`. You give it a callback; it gives
+you information. The information that the callback receives is a list of issues
+that the page caused:
+
+```js
+const observer = new ReportingObserver((reports, observer) => {
+  for (const report of reports) {
+    // → report.type === 'deprecation'
+    // → report.url === 'https://reporting-observer-api-demo.glitch.me'
+    // → report.body.id === 'XMLHttpRequestSynchronousInNonWorkerOutsideBeforeUnload'
+    // → report.body.message === 'Synchronous XMLHttpRequest is deprecated...'
+    // → report.body.lineNumber === 11
+    // → report.body.columnNumber === 22
+    // → report.body.sourceFile === 'https://reporting-observer-api-demo.glitch.me'
+    // → report.body.anticipatedRemoval === <JS_DATE_STR> or null
+  }
+});
+
+observer.observe();
+```
+
+### Filtered reports
+
+Reports can be pre-filtered to only observe certain report types. Right now,
+there are two report types: `'deprecation'` and `'intervention'`.
+
+```js
+const observer = new ReportingObserver((reports, observer) => {
+  ...
+}, {types: ['deprecation']});
+```
+
+### Buffered reports
+
+Use the The `buffered: true` option when you want to see the reports that were
+generated before the observer instance was created:
+
+```js
+const observer = new ReportingObserver((reports, observer) => {
+  ...
+}, {types: ['intervention'], buffered: true});
+```
+
+This option is great for situations like lazy-loading a library that uses a
+`ReportingObserver`. The observer gets added late, but you don't miss out on
+anything that happened earlier in the page load.
+
+### Stop observing
+
+Stop observing using the `disconnect()` method:
+
+```js
+observer.disconnect();
+```
+
+## Examples
+
+### Report browser interventions to an analytics provider
+
+```js
+const observer = new ReportingObserver((reports, observer) => {
+  for (const report of reports) {
+    sendReportToAnalytics(JSON.stringify(report.body));
+  }
+}, {types: ['intervention'], buffered: true});
+
+observer.observe();
+```
+
+### Be notified when APIs are going to be removed
+
+```js
+const observer = new ReportingObserver((reports, observer) => {
+  for (const report of reports) {
+    if (report.type === 'deprecation') {
+      sendToBackend(`Using a deprecated API in ${report.body.sourceFile} which will be
+                     removed on ${report.body.anticipatedRemoval}. Info: ${report.body.message}`);
+    }
+  }
+});
+
+observer.observe();
+```
+
+## Conclusion
+
+`ReportingObserver` gives you an additional way for discovering and monitoring
+potential issues in your web app. It's even a useful tool for understanding the
+health of your code base (or lack thereof). Send reports to a backend, know
```

Comment:
s/backend/back end/

Reply:
@philipwalton style guide convention is to write it as one word: https://developers.google.com/style/word-list#backend

Reply:
Fixed.